services/user/Fractals/service/src/tables.rs

```diff
@@ -0,0 +1,573 @@
+            let current_time = TransactSvc::call().currentBlock().time.seconds();
+
+            let withdrawable = self.withdrawable_amount();
+            if withdrawable == 0 {
+                abort_message("nothing to withdraw");
+            }
+
+            self.reward_balance = self.reward_balance.saturating_sub(withdrawable);
+            self.reward_start_time = current_time;
+
+            let reward_wait_period = Fractal::get_assert(self.fractal).reward_wait_period;
+
+            self.reward_wait = if self.reward_balance == 0 {
+                reward_wait_period
+            } else {
+                let original_balance = self.reward_balance + withdrawable;
+                let new_period = if original_balance == 0 {
+                    reward_wait_period as f64
+                } else {
+                    (self.reward_wait as f64 * self.reward_balance as f64) / original_balance as f64
+                };
+                new_period.ceil() as u32
+            };
+
+            self.save();
+        }
+
+        pub fn add(fractal: AccountNumber, account: AccountNumber, status: MemberStatus) -> Self {
+            let new_instance = Self::new(fractal, account, status);
+            new_instance.save();
+
+            new_instance.initialize_score(EvalType::Repuation);
+            new_instance.initialize_score(EvalType::Favor);
+
+            new_instance
+        }
+
+        pub fn initialize_score(&self, eval_type: EvalType) {
+            Score::add(self.fractal, eval_type, self.account);
+        }
+
+        pub fn get(fractal: AccountNumber, account: AccountNumber) -> Option<Member> {
+            let table = MemberTable::new();
+            table.get_index_pk().get(&(fractal, account))
+        }
+
+        pub fn get_assert(fractal: AccountNumber, account: AccountNumber) -> Self {
+            check_some(Self::get(fractal, account), "member does not exist")
+        }
+
+        pub fn save(&self) {
+            let table = MemberTable::new();
+            table.put(&self).expect("failed to save");
+        }
+    }
+
+    #[derive(PartialEq, Clone)]
+    pub enum EvalType {
+        Repuation = 1,
+        Favor = 2,
+    }
+
+    pub type EvalTypeU32 = u32;
+
+    impl From<EvalTypeU32> for EvalType {
+        fn from(eval_type: EvalTypeU32) -> Self {
+            match eval_type {
+                1 => EvalType::Repuation,
+                2 => EvalType::Favor,
+                _ => abort_message("invalid evaluation type"),
+            }
+        }
+    }
+
+    impl From<EvalType> for u32 {
+        fn from(eval_type: EvalType) -> u32 {
+            eval_type as u32
+        }
+    }
+
+    #[table(name = "EvaluationInstanceTable", index = 2)]
+    #[derive(Default, Fracpack, ToSchema, SimpleObject, Serialize, Deserialize, Debug)]
+    pub struct EvaluationInstance {
+        pub fractal: AccountNumber,
+        pub eval_type: EvalTypeU32,
+        pub interval: u32,
+        pub evaluation_id: Option<u32>,
```

Comment:
I think this should just be u32, and `get_or_create` should actually create the evaluation, not just add a record to this table with no ID.